wadsrc/static/zscript/actors/inventory_util.zs

```diff
@@ -163,17 +159,18 @@ extend class Actor
 
 		// Do not take ammo if the "no take infinite/take as ammo depletion" flag is set
 		// and infinite ammo is on
+		// Alternatively do not take customInventory items if sv_infiniteinventory is true
 		if (notakeinfinite &&
-		(sv_infiniteammo || (player && FindInventory('PowerInfiniteAmmo', true))) && (item is 'Ammo'))
+		(sv_infiniteammo || (player && FindInventory('PowerInfiniteAmmo', true))) && (item is 'Ammo')
+		|| sv_infiniteinventory && item is 'CustomInventory')
```

Comment:
I was wrong. TakeInventory is never called if sv_infiniteinventory is true, so this part alone should be reverted. Otherwise this could cause some problems.

Sorry about that.